core/src/main/scala/akka/persistence/r2dbc/journal/JournalDao.scala

```diff
@@ -87,6 +87,8 @@ private[r2dbc] class JournalDao(journalSettings: R2dbcSettings, connectionFactor
 
   private val journalTable = journalSettings.journalTableWithSchema
 
+  // Note: wasn't the thing that we can't use transaction_timestamp() because start of transaction rather than actual commit
```

Comment:
There is no way to insert the timestamp after commit, it's always when the transaction or statement started. This is one of the main reasons  why we need the backtracking and deduplication.